### ISSUE TITLE
Bootstrapping vis-plug

### ISSUE THREAD

**Issue by unrealapex** [erf/vis-plug#35]
vis-plug currently has no means of bootstrapping automatically. Adding an example of how to achieve this in the README would be helpful.

**Comment by erf** [comment]
I mention in the `README.md` that:

`git clone vis-plug and require it in your visrc.`

As you would normally require plugins using `vis`.

**Comment by erf** [comment]
If you have a suggestion on how to improve this. Please

**Comment by unrealapex** [comment]
> `git clone vis-plug and require it in your visrc.`

I suggest having a snippet that does this. If you would like, I can submit a PR for this:
```lua
local visplugpath = os.getenv("XDG_CONFIG_HOME") .. "/vis/vis-plug"
local file_exists = function(path)
        local file = io.open(path)
        if not file then return false end
        file:close()
        return true
end


if not file_exists(visplugpath)
then
  print("vis-plug not found, installing...")
  os.execute("git clone --filter=blob:none https://github.com/erf/vis-plug " .. visplugpath)
  require("vis-plug")
end
```

**Comment by erf** [comment]
This would basically be a another plugin just to fetch `vis-plug` ? 

Maybe i feel it's a bit overkill. Also maybe the vis config home can be different and needs more checks. You'd also probably want to add `vis-plug` in `/vis/plugins/vis-plug`.

Maybe you could add a separate plugin to boot strap it - or how do you imagine to use this?

**Comment by unrealapex** [comment]
> how do you image to use this?

I would put this in my visrc so I could ensure vis-plug is installed before any plugins are loaded.

I should probably change `visplugpath` to something more sensible.

**Comment by erf** [comment]
I think i might prefer a optional bash script to automatically clone the repo to the right place as not to pollute the visrc file too much. It's usually just a one time thing.

**Comment by erf** [comment]
When i think about it a separate bash script would require you to either clone or copy the whole file. Rather a simple one-liner you could just copy would work best i think.

Regarding your PR - you'd want to place `vis-plug` in one of the `package.path` that `vis` searches for. `XDG_CONFIG_HOME` is one of those, but there are many as described [here](https://github.com/martanne/vis/blob/c0d083f4ca827d83c6b8beafd52a0ea95ece73da/vis-lua.c#L3162).

Although, we could ask the user to first set `$XDG_CONFIG_HOME` and only check for that and perhaps also include `$HOME/.config/vis`.

**Comment by erf** [comment]
Solved via ea2155ec45684ebf091dcd9041e0588bfdaee2a2

**Comment by sewnie** [comment]
It would be nice if this bootstrapper could be documented. I do not want to run a one-liner, and would want it to be seamless within my editor.
